diff --git a/Kconfig b/Kconfig
index fc4c9f4..673472c 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1394,7 +1394,6 @@ config BPF
 menuconfig EXPERT
 	bool "Configure standard kernel features (expert users)"
 	# Unhide debug options, to make the on-by-default options visible
-	select DEBUG_KERNEL
 	help
 	  This option allows certain base kernel options and settings
 	  to be disabled or tweaked. This is for specialized
